viscy/utils/mp_utils.py

```diff
@@ -247,43 +247,85 @@ def mp_get_val_stats(fn_args, workers):
     """
     with ProcessPoolExecutor(workers) as ex:
         # can't use map directly as it works only with single arg functions
-        res = ex.map(get_val_stats, fn_args)
+        res = ex.map(get_val_stats, *zip(*fn_args))
     return list(res)
 
 
-def get_val_stats(sample_values):
+def get_val_stats(sample_values, percentiles=(25.0, 75.0)):
     """
     Computes the statistics of a numpy array and returns a dictionary
     of metadata corresponding to input sample values.
 
-    :param list(float) sample_values: List of sample values at respective
-                                        indices
-    :return dict meta_row: Dict with intensity data for image
+    Parameters
+    ----------
+    sample_values : array_like
+        List of sample values at respective indices
+    percentiles : tuple of float, optional
+        Lower and upper percentiles to compute, by default (25.0, 75.0)
+        
+    Returns
+    -------
+    dict
+        Dict with intensity data for image including mean, std, median, iqr,
+        percentile bounds and range
     """
-
+    percentile_vals = np.nanpercentile(sample_values, percentiles)
```

Comment:
This call can also compute median (50%) and IQR (25%, 75%) in one sorting operation.